components/TestimoniesTable/TestimoniesTable.js

```diff
@@ -15,14 +15,19 @@ const TestimonyRow = ({ testimony }) => {
   const senatorId = testimony.senatorId
   const representativeId = testimony.representativeId
 
+  const x = usePublicProfile(testimony.authorUid)
+  const authorPublic = x.result?.public
+  const loading = x.loading
+  const error = x.error
+
   return (
     <tr>
       <td>{testimony.billId}</td>
       <td>{testimony.position}</td>
       <td>
         {testimony.authorDisplayName == null ? (
-          "(blank)"
-        ) : (
+          <>(blank)</>
+        ) : authorPublic ? (
           <Button
```

Comment:
We also have link to the user profile from the home page testimony table (maybe elsewhere, I'm not 100% sure). Can we create a ProfileButton component and share it between those places? It would take the uid as a prop, call `usePublicProfile`, and render either the button or text

Reply:
Right now the name isn't being saved in the profile if the user signs up with an email address, so we need to pass the user name as well - it is stored with the testimony.  We can remove passing the name we resolve the missing names issue.

![image](https://user-images.githubusercontent.com/67333843/161850666-808374e9-b8cb-4976-a47d-ba4dd72ce0b0.png)